matchms/utils.py

```diff
@@ -73,3 +73,25 @@ def _load_key_conversions(file: str, source: str, target: str) -> dict:
 def fingerprint_export_warning(spectrums: List[SpectrumType]):
     if any(x.get("fingerprint") is not None for x in spectrums):
         logger.warning("fingerprint found but will not be written to file.")
+
+
+def return_non_existing_file_name(file_name):
+    """Checks if a path already exists, otherwise creates a new filename with (1)"""
+    if not os.path.exists(file_name):
+        return file_name
+    print(f"The file name already exists: {file_name}")
+    file_name_base, file_extension = os.path.splitext(file_name)
+    i = 1
+    new_file_name = f"{file_name_base}({i}){file_extension}"
+    while os.path.exists(new_file_name):
+        i += 1
+        new_file_name = f"{file_name_base}({i}){file_extension}"
+    print(f"Instead the file will be stored in {new_file_name}")
+    return new_file_name
+
+
+def create_dir_if_missing(folder):
+    """Checks if the folder already exists otherwise create the folder"""
+    if not os.path.isdir(folder):
+        assert not os.path.isfile(folder), f"{folder} is expected to be a folder but a file was found"
+        os.mkdir(folder)
```

Comment:
The regular mkdir function already throws an exception if the folder exists - see https://docs.python.org/3/library/os.html#os.mkdir

Reply:
My issue with os.mkdir, was actually that it throws an exception when the folder with that exact name already exists. When running the pipeline myself, I did often rerun previous run pipelines so sometimes the folder was already created and sometimes it was not yet there, making the code crash. 

But the assert statement to check if it is a file is not needed indeed. 

Not sure if it should be added to matchms, but, since I found it useful from a user perspective, I thought it might be more user friendly for other users as well.

Reply:
I mean you can use try - catch to deal with the exception - an assert is actually going to do exactly the same, and also assert is very slow and only supposed to be used in testing and debugging, so I'd like to keep this out of running production code.